src/Bicep.Core/TypeSystem/DeclaredTypeManager.cs

```diff
@@ -914,12 +914,12 @@ private TypeSymbol GetResourceTypeFromString(TextSpan span, string stringContent
                     throw new InvalidOperationException($"qualifiedTypeReference is null");
                 }
 
-                if (binder.NamespaceResolver.TryGetResourceType(typeReference, typeGenerationFlags) is { } resourceType)
-                {
-                    return resourceType;
-                }
-
-                return ErrorType.Create(DiagnosticBuilder.ForPosition(span).InvalidResourceType());
+                var resourceTypes = binder.NamespaceResolver.GetMatchingResourceTypes(typeReference, typeGenerationFlags);
+                return resourceTypes.Length switch {
+                    0 => ErrorType.Create( DiagnosticBuilder.ForPosition(span).InvalidResourceType()),
+                    1 => resourceTypes[0],
+                    _ => ErrorType.Create(DiagnosticBuilder.ForPosition(span).AmbiguousResourceTypeBetweenImports(typeReference.FormatName(), resourceTypes.Select(x => x.DeclaringNamespace.Name))),
```

Comment:
I think the enumerable of namespaces will need to be sorted to avoid test failures like https://github.com/Azure/bicep/runs/7853142549?check_suite_focus=true#step:6:125